pygac/reader.py

```diff
@@ -0,0 +1,894 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+# Copyright (c) 2014, 2019 Pygac Developers
+
+# Author(s):
+
+#   Martin Raspaud <martin.raspaud@smhi.se>
+
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Generic reader for GAC and LAC data.
+
+Can't be used as is, has to be subclassed to add specific read functions.
+"""
+from abc import ABCMeta, abstractmethod, abstractproperty
+import logging
+import numpy as np
+import os
+import six
+import types
+
+from pygac import (CONFIG_FILE, centered_modulus,
+                   get_absolute_azimuth_angle_diff)
+try:
+    import ConfigParser
+except ImportError:
+    import configparser as ConfigParser
+
+from pyorbital.orbital import Orbital
+from pyorbital import astronomy
+import datetime
+from pygac.calibration import calibrate_solar, calibrate_thermal
+
+LOG = logging.getLogger(__name__)
+
+# rpy values from
+# here:http://yyy.rsmas.miami.edu/groups/rrsl/pathfinder/Processing/proc_app_a.html
+rpy_coeffs = {
+    'noaa7':  {'roll':  0.000,
+               'pitch': 0.000,
+               'yaw':   0.000,
+               },
+    'noaa9':  {'roll':  0.000,
+               'pitch': 0.0025,
+               'yaw':   0.000,
+               },
```

Comment:
@sfinkens could this be a problem ?

Reply:
Indeed, that's the reason!

Reply:
So now we just have to decide if it's making the data better or not, right ? or should I just remove this ?